addons/account/models/account_bank_statement.py

```diff
@@ -682,7 +682,7 @@ def _prepare_counterpart_move_line_vals(self, counterpart_vals, move_line=None):
             if balance_to_reconcile is None else balance_to_reconcile
 
         if 'currency_id' in counterpart_vals:
-            currency_id = counterpart_vals['currency_id'] or foreign_currency.id
+            currency_id = counterpart_vals['currency_id'] or company_currency.id
```

Comment:
I'm pretty sure `foreign_currency` is correct... It should be at least. Let me check exactly what your issue is...